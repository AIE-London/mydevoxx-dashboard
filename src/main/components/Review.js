import React, { Component } from "react";
import styled from "styled-components";
import { Col, Row } from "react-flexbox-grid";

/**
 *  Styled Components
 */

const ReviewText = styled.p`
  font-family: Helvetica;
  font-weight: 500;
  font-size: 1em;
  margin: 0;
  opacity: 0.6;
`;

const Name = styled.h3`
  font-family: Helvetica;
  font-weight: 500;
  font-size: 1.25em;
  margin: 0;
  opacity: 0.6;
`;

const ProfileImg = styled.img`
  border-radius: 50%;
  height: 30px;
  width: 30px;
  border: solid white 3px;
`;

class Review extends Component {
  render() {
    try {
      let { image, name, comment } = this.props.review;
      return (
        <Row start="xs">
          <Col>
            {image && <ProfileImg src={image} />}
          </Col>
          <Col xs>
            <Name>{name}</Name>
            <ReviewText>{comment}</ReviewText>
          </Col>
        </Row>
      );
    } catch (error) {
      return <Row />;
    }
  }
}

export default Review;
