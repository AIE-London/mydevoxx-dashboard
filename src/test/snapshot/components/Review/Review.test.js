/**
 * Created by dan on 05/04/2017.
 */
import React from 'react';
import ReactDOM from 'react-dom';
import Review from '../../../../main/components/Review';
import renderer from 'react-test-renderer';
import testImage from '../../images/test-image.jpeg';

const fullyPopReview = {
  name: 'Test User',
  comment: 'Great session, thanks for organising. Looking forward to the next one!',
  image: testImage
};

const emptyReview = {

};

const reviewWithoutImage = {
  name: 'Test User 3',
  comment: '3 Great session, thanks for organising. Looking forward to the next one!'
};

test('Review Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Review />, div);
  });
});

test('Review Component - Fully populated review', () => {
  const tree = renderer.create(
    <Review review={fullyPopReview} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Review Component - Empty Review', () => {
  const tree = renderer.create(
    <Review review={emptyReview} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Review Component - Missing Image', () => {
  const tree = renderer.create(
    <Review review={reviewWithoutImage} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});