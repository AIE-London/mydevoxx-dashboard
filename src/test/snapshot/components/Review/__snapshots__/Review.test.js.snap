exports[`test Review Component - Empty Review 1`] = `
<div
  className="row start-xs">
  <div
    className="" />
  <div
    className="col-xs">
    <h3
      className="jPwRym" />
    <p
      className="ZxTsc" />
  </div>
</div>
`;

exports[`test Review Component - Fully populated review 1`] = `
<div
  className="row start-xs">
  <div
    className="">
    <img
      className="imhmHM"
      src="test-image.jpeg" />
  </div>
  <div
    className="col-xs">
    <h3
      className="jPwRym">
      Test User
    </h3>
    <p
      className="ZxTsc">
      Great session, thanks for organising. Looking forward to the next one!
    </p>
  </div>
</div>
`;

exports[`test Review Component - Missing Image 1`] = `
<div
  className="row start-xs">
  <div
    className="" />
  <div
    className="col-xs">
    <h3
      className="jPwRym">
      Test User 3
    </h3>
    <p
      className="ZxTsc">
      3 Great session, thanks for organising. Looking forward to the next one!
    </p>
  </div>
</div>
`;
